pkg/controller/pd_control.go

```diff
@@ -21,14 +21,14 @@ import (
 // GetPDClient gets the pd client from the TidbCluster
 func GetPDClient(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) pdapi.PDClient {
 	if tc.IsHeterogeneous() {
-		return pdControl.GetPDClient(pdapi.Namespace(tc.GetNamespace()), tc.Spec.Cluster.Name, tc.IsTLSClusterEnabled())
+		return pdControl.GetPDClient(pdapi.Namespace(tc.Spec.Cluster.Namespace), tc.Spec.Cluster.Name, tc.Spec.Cluster.Domain, tc.IsTLSClusterEnabled())
```

Comment:
`tc.Spec.Cluster.Namespace` this is optional, if you want to use this, you have to set default values for it in https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/defaulting/tidbcluster.go#L58

Reply:
And this comment is not addressed.